haystack/nodes/file_converter/pdf.py

```diff
@@ -169,15 +178,28 @@ def _read_pdf(self, file_path: Path, layout: bool, encoding: Optional[str] = Non
                        the content stream order.
         :param encoding: Encoding that overwrites self.encoding and will be passed as `-enc` parameter to `pdftotext`.
                          (See list of available encodings by running `pdftotext -listenc` in the terminal)
+        :param start_page: The page number where to start the conversion
+        :param end_page: The page number where to end the conversion.
         """
         if not encoding:
             encoding = self.encoding
 
         command = ["pdftotext", "-enc", str(encoding), "-layout" if layout else "-raw", str(file_path), "-"]
+
+        start_page = start_page or 1
+        command.insert(3, "-f")
+        command.insert(4, str(start_page))
+
+        if end_page is not None:
+            command.insert(5, "-l")
+            command.insert(6, str(end_page))
```

Comment:
Similarily if we can avoid using `inserts` here I think that would be best.

Reply:
I'll change the list creation process a little bit so.

Reply:
Sounds good, happy to have the list creation process change!